From 8739bea3143d1194b7f9e81546fa586bbfcfc547 Mon Sep 17 00:00:00 2001
From: Phil Edworthy <phil.edworthy@renesas.com>
Date: Mon, 10 Dec 2012 16:12:27 +0000
Subject: [PATCH 087/215] video: rcarfb: Add ability to set resolution on the
 cmd line

For example, pass "video=rcarfb0:800x600-16" to set 800x600 at
16bpp.

Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 drivers/video/rcarfb.c |   28 +++++++++++++++++++++++++++-
 1 file changed, 27 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rcarfb.c b/drivers/video/rcarfb.c
index 388923b..4aba68a 100644
--- a/drivers/video/rcarfb.c
+++ b/drivers/video/rcarfb.c
@@ -964,7 +964,33 @@ static struct platform_driver rcar_du_driver = {
 	.remove		= rcar_du_remove,
 };
 
-module_platform_driver(rcar_du_driver);
+static int __init rcar_du_init(void)
+{
+	char *option;
+	char *this_opt;
+
+	if (fb_get_options("rcarfb0", &option))
+		return -ENODEV;
+
+	while ((this_opt = strsep(&option, ",")) != NULL)
+		mode_option[0] = this_opt;
+
+	if (fb_get_options("rcarfb1", &option))
+		return -ENODEV;
+
+	while ((this_opt = strsep(&option, ",")) != NULL)
+		mode_option[1] = this_opt;
+
+	return platform_driver_register(&rcar_du_driver);
+}
+
+static void __exit rcar_du_exit(void)
+{
+	platform_driver_unregister(&rcar_du_driver);
+}
+
+module_init(rcar_du_init);
+module_exit(rcar_du_exit);
 
 MODULE_DESCRIPTION("R-Car Display Unit Framebuffer driver");
 MODULE_LICENSE("GPL v2");
-- 
1.7.9.5

